### DISCUSSION TITLE
Refitter & Apizr

### DISCUSSION THREAD

**Discussion by JeremyBP** [christianhelle/refitter#404] [Ideas]
Hi, as I'm about to release the v6 of [Apizr](https://www.apizr.net), I'm thinking about droping my custom NSwag tool that generate a Refit interface with Apizr attributes + some registration helpers (you'll find this [documentation article](https://www.apizr.net/articles/tools_nswag.html) about it to understand what it does).

My nswag tool is not the main value on which Apizr is working on, and I don't want to maintain such a tool while Refitter do it so well.
So I was wondering if instead of maintaining my current nswag tool or recreating the wheel again with a new tool based on Refitter, maybe I could push some PRs here to let the user choose to activate some kind of Apizr specific things, right from .refitter file.

It would be first about including an `[RequestOptions] IApizrRequestOptions options` parameter to api methods, like you did with cancellation tokens. It could be then many more things we could talk about.

Anyway, my goal is to not create a new Refitter or something but to stay focused on Apizr core features and I think Refitter & Apizr  could work together at some point.

Let me know.
Jeremy

**Comment by christianhelle** [comment]
Thanks for taking the time to bring this up @JeremyBP 

I haven't heard of Apizr until now, but it looks interesting. Great stuff!

Most of the features in Refitter are community contributions and I'm very open to all new features as long as they don't break existing functionality. Just come with the pull requests and I'll make sure we get them in so it works for your tool

**Comment by christianhelle** [comment]
Refitter is also available as [a library](https://www.nuget.org/packages/Refitter.Core/1.0.2) in case you need to use it directly from code, instead of launching the Refitter CLI tool process. I use the core library in my Visual Studio extension called [REST API Client Code Generator](https://github.com/christianhelle/apiclientcodegen)

**Comment by JeremyBP** [comment]
Thanks for code merge \o/
I think it's a nice match between Refitter & Apizr.
Do you want me to update your Docfx md files in another PR about Apizr integration? or do I let you do it as the owner?

**Reply by christianhelle** [reply]
I would really appreciate it if you could help with the docs as I'm not very familiar with Apizr yet. 

Just some basic information in the READMEs and maybe a DocFX article with links to the official Apizr docs would be awesome

**Comment by JeremyBP** [comment]
I just pushed PR #430 to complete the puzzle with command line capabilities.

Now I'll work on some doc updates about Apizr (readme + a dedicated basic settings article)

Then I'll give a try to #417 implementation

**Reply by christianhelle** [reply]
Awesome! I'll review these tonight when I get access to a computer. I'm at the beach with only my phone right now

**Reply by christianhelle** [reply]
#430 was small enough that I could actually just review it on my phone while waiting for lunch 😃

**Comment by JeremyBP** [comment]
Well, you know me, I'm the guy that won't let you chill at the beach ^^
Sorry about my PR spams... Anyway, I hope it's a nice feeling to see someone else working on your project while your in holidays 🏊 My pleasure haha 😄 
That said, I just pushed the #431 to include the missing xml doc param for the Apizr request options parameter. Test covered & 400+ all green.

In the other hand, I finalized all doc articles and readmes refreshing, but I'll push it in another PR to keep things separated.
So yes, another PR should follow for doc updates after that one merged.
Then I think we'll be done with that #404 hopefully :)

Next week will be the #417 as planned and then I'll be on holidays too 😃

**Reply by christianhelle** [reply]
> Well, you know me, I'm the guy that won't let you chill at the beach ^^

That's my problem that I even look at Github while at the beach. I could have just ignored notifications

> Sorry about my PR spams... Anyway, I hope it's a nice feeling to see someone else working on your project while your in holidays 🏊 My pleasure haha 😄

It's cool and I appreciate the work you do on the project 😃 

> In the other hand, I finalized all doc articles and readmes refreshing, but I'll push it in another PR to keep things separated.
> So yes, another PR should follow for doc updates after that one merged.
> Then I think we'll be done with that #404 hopefully :)

I'm looking forward and thanks for putting so much effort into this

> Next week will be the #417 as planned and then I'll be on holidays too 😃

Remember to take a "real" holiday and not just stare at another screen all the time 😁

**Comment by JeremyBP** [comment]
That one is now implemented 🚀

**Reply by christianhelle** [reply]
> That one is now implemented 🚀

And published to [nuget.org as a v1.2.0 (preview)](https://www.nuget.org/packages/Refitter/1.2.0-preview.52)
